File: geo-building-extractor/backend/map-service/src/geo_utils.py
def calculate_area(rectangle):
    """Calculate the area of a rectangle given its coordinates."""
    x1, y1, x2, y2 = rectangle
    return abs((x2 - x1) * (y2 - y1))

def rectangle_to_geojson(rectangle):
    """Convert rectangle coordinates to GeoJSON format."""
    x1, y1, x2, y2 = rectangle
    geojson = {
        "type": "Feature",
        "geometry": {
            "type": "Polygon",
            "coordinates": [[
                [x1, y1],
                [x2, y1],
                [x2, y2],
                [x1, y2],
                [x1, y1]
            ]]
        },
        "properties": {}
    }
    return geojson

def extract_building_footprints(rectangle):
    """Placeholder function to extract building footprints within a rectangle."""
    # This function would interact with a GIS database or API to retrieve building footprints
    return []  # Return an empty list as a placeholder

def validate_rectangle(rectangle):
    """Validate the rectangle coordinates."""
    if len(rectangle) != 4:
        raise ValueError("Rectangle must have four coordinates.")
    x1, y1, x2, y2 = rectangle
    if x1 >= x2 or y1 >= y2:
        raise ValueError("Invalid rectangle coordinates.")

File: map-extraction-service - claude - monolothic/frontend/src/utils/geoUtils.ts
// This file contains utility functions for geo-related operations.

export const createGeoJSONRectangle = (coordinates: [number, number][]) => {
    return {
        type: "Feature",
        geometry: {
            type: "Polygon",
            coordinates: [coordinates],
        },
        properties: {},
    };
};

export const calculateArea = (coordinates: [number, number][]) => {
    // Simple calculation for area in square meters based on coordinates
    // This is a placeholder and should be replaced with a proper area calculation
    const length = Math.abs(coordinates[0][0] - coordinates[2][0]);
    const width = Math.abs(coordinates[0][1] - coordinates[1][1]);
    return length * width; // This is a simplistic approach
};

export const extractBuildingFootprints = async (rectangle: any) => {
    // Placeholder for API call to extract building footprints
    // This function should call the backend service to get the footprints based on the rectangle
    const response = await fetch('/api/geo/extract', {
        method: 'POST',
        headers: {
            'Content-Type': 'application/json',
        },
        body: JSON.stringify(rectangle),
    });
    return response.json();
};